src/models/User.js

```diff
@@ -4,91 +4,101 @@ const jwt = require('jsonwebtoken')
-                    )
-                }
-            },
-        },
-        phoneNumber: {
-            type: String,
-            trim: true,
-            required: true,
-            validate(value) {
-                if (!validator.isMobileNumber(value)) {
-                    throw new Error('Invalid Mobile Number')
-                }
-            },
-        },
-        tokens: [
+
+const userSchema = mongoose.Schema({
+    name:{
+        type:String,
+        trim:true,
+        required:true
+    },
+    email:{
+        type:String,
+        trim:true,
+        required:true,
+        unique:true,
+        validate(value)
+        {
+            if(!validator.isEmail(value))
+            {
+                throw new Error('Email is invalid')
+            }
+        }
+    },
+    password:{
+        type:String,
+        trim:true,
+        required:true,
+        minlength:8,
+        validate(value)
+        {
+            var strength=0
+            if(value.match(/[a-z]+/)){
+                strength+=1;
+            }
+            if(value.match(/[A-Z]+/)){
+                strength+=1;
+            }
+            if(value.match(/[0-9]+/)){
+            strength+=1;}
+            if(value.match(/[#$&!@]+/)){
+                strength+=1;
+            }
+            if(strength<4)
             {
-                token: {
-                    type: String,
-                    required: true,
-                },
-            },
-        ],
+                throw new Error('The password must contain a mix of uppercase and lowercase alphabets along with numbers and special chacracters')
+            }
+        }
     },
-    {
-        timestamps: true,
+    phoneNumber:{
+        type:String,
+        trim:true,
+        required:true,
+        validate(value)
+        {
+            if(!validator.isMobilePhone(value))
+            {
+                throw new Error('Invalid Mobile Number')
+            }
+        }
     }
-)
+    /*tokens:[
+        {
+            token:{
+                type:String,
+                required:true
+            }
 
-userSchema.methods.generateAuthToken = async function () {
-    const user = this
-    const token = jwt.sign({ _id: user._id.toString() }, process.env.JWT_SECRET)
-    user.tokens = user.tokens.concat({ token })
-    await user.save()
-    return token
+        }
+    ]*/
 }
+/*{
+    timestamps:true
+}*/);
+
+
+
+// static method to login user
+userSchema.statics.login = async function(email, password) {
+    const user = await this.findOne({ email });
+    if (user) {
+      const auth = await bcrypt.compare(password, user.password);
+      if (auth) {
+        return user;
+      }
+      throw Error('incorrect password');
 
-userSchema.pre('save', async function (next) {
-    const user = this
-    if (user.isModified('password')) {
-        user.passsword = await bcrypt.hash(user.passsword, 8)
     }
-    next()
-})
+    throw Error('incorrect email');
```

Comment:
Don't write incorrect emai This gives information to the use about the error.Just write `Invalid Credentials`

Reply:
Alright, I shall make this change